extensions/prometheus/prometheus.go

```diff
@@ -0,0 +1,140 @@
+package hogprom
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/exaring/hoglet"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	namespace = "hoglet"
+)
+
+// WithPrometheusMetrics returns a [hoglet.BreakerMiddleware] that registers prometheus metrics for the circuit breaker.
```

Comment:
nit: the whole selling point of hoglet is somewhat decoupling the circuit and the break, we should be clear with the namings here too. 
```suggestion
// WithPrometheusMetrics returns a [hoglet.BreakerMiddleware] that registers prometheus metrics for the circuit.
```